### ISSUE TITLE
Error when Installing RAM

### ISSUE THREAD

**Issue by OldCaptainWZJ** [IDEA-Research/Grounded-Segment-Anything#457]
When building the environment locally, after running command 'pip install -e ./recognize-anything/', following error occured, seems like something wrong with setup.py:

(myconda) root@ZDEObv:~/Grounded-Segment-Anything# pip install -e ./recognize-anything/
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Obtaining file:///root/Grounded-Segment-Anything/recognize-anything
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [41 lines of output]
      Traceback (most recent call last):
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/packaging/requirements.py", line 102, in __init__
          req = REQUIREMENT.parseString(requirement_string)
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 1654, in parseString
          raise exc
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 1644, in parseString
          loc, tokens = self._parse( instring, 0 )
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 1402, in _parseNoCache
          loc,tokens = self.parseImpl( instring, preloc, doActions )
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 3417, in parseImpl
          loc, exprtokens = e._parse( instring, loc, doActions )
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 1406, in _parseNoCache
          loc,tokens = self.parseImpl( instring, preloc, doActions )
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/pyparsing.py", line 3205, in parseImpl
          raise ParseException(instring, loc, self.errmsg, self)
      pkg_resources._vendor.pyparsing.ParseException: Expected stringEnd (at char 4), (line:1, col:5)
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/root/Grounded-Segment-Anything/recognize-anything/setup.py", line 2, in <module>
          setuptools.setup()
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/setuptools/__init__.py", line 154, in setup
          _install_setup_requires(attrs)
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/setuptools/__init__.py", line 146, in _install_setup_requires
          dist.parse_config_files(ignore_option_errors=True)
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/setuptools/dist.py", line 807, in parse_config_files
          self._finalize_requires()
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/setuptools/dist.py", line 534, in _finalize_requires
          self._move_install_requirements_markers()
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/setuptools/dist.py", line 573, in _move_install_requirements_markers
          inst_reqs = list(pkg_resources.parse_requirements(spec_inst_reqs))
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3099, in parse_requirements
          yield Requirement(line)
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3109, in __init__
          super(Requirement, self).__init__(requirement_string)
        File "/root/miniconda3/envs/myconda/lib/python3.10/site-packages/pkg_resources/_vendor/packaging/requirements.py", line 104, in __init__
          raise InvalidRequirement(
      pkg_resources.extern.packaging.requirements.InvalidRequirement: Parse error at "': requir'": Expected stringEnd
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by coderczhhh** [comment]
I met the same problem

**Comment by zeanfa** [comment]
https://github.com/xinyu1205/recognize-anything/issues/107#issue-1973305162

TL;DR
pip install setuptools --upgrade

**Comment by OldCaptainWZJ** [comment]
> [xinyu1205/recognize-anything#107 (comment)](https://github.com/xinyu1205/recognize-anything/issues/107#issue-1973305162)
> 
> TL;DR pip install --upgrade setuptools

Thanks, this works
